smore/pagination/models.py

```diff
@@ -0,0 +1,124 @@
+import abc
+import math
+import collections
+
+import six
+
+
+class BasePage(six.with_metaclass(abc.ABCMeta, collections.Sequence)):
```

Comment:
`marshmallow.compat` already provides a `with_metaclass` function, so `six` shouldn't be necessary here.